data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/configuration/MongoDBSourceConfig.java

```diff
@@ -15,8 +16,8 @@ public class MongoDBSourceConfig {
     private static final String DEFAULT_READ_PREFERENCE = "primaryPreferred";
     private static final Boolean DEFAULT_DIRECT_CONNECT = true;
     private static final Duration DEFAULT_ACKNOWLEDGEMENT_SET_TIMEOUT = Duration.ofHours(2);
-    @JsonProperty("host")
-    private @NotNull String host;
+    @JsonProperty("hosts")
+    private @NotNull String[] hosts;
```

Comment:
Any reason String[] is used instead of List<String>. Does it affect anything?

Reply:
This will be reverted to `String host`. I will have this in followup CR.